autosklearn/ensemble_building/builder.py

```diff
@@ -169,6 +169,8 @@ def __init__(
         self.validation_performance_ = np.inf
 
         # Data we may need
+        # TODO: The test data is needlessly loaded but automl_common has no concept of
```

Comment:
Do you mean train data is needlessly loaded?

Reply:
Nope, the `X_ensemble` and `y_ensemble` are loaded on demand as we like. Specifically for `X_ensemble` this means we only load it if a metrics `_needs_X`.

However for `_X_test` and `_y_test`, these are already loaded once we `load_datamanager()` so we may as well store them. The end result is that `_X_test` is loaded if it's in the Datamanger, whether we need it or not.

https://github.com/automl/auto-sklearn/blob/c07856e81b82584c2a479d0aabdc2afe745d18cf/autosklearn/ensemble_building/builder.py#L171-L178